src/document/reducer.ts

```diff
@@ -118,11 +120,15 @@ function updateOperations<T extends Document>(
         }
 
         nextIndex = action.index;
+        operationId = action.id;
+    } else {
+        operationId = uuid();
```

Comment:
This makes the reducer non pure which might be a problem but let's leave it like this for now to make the transition easier.